### DISCUSSION TITLE
Asciidoctor Gem Not Found

### DISCUSSION THREAD

**Discussion by FrankAllotta** [usnistgov/macos_security#171] [Q&A] [answered]
When I try to run the generate_guidance.py it shows the following messages:

```
If you would like to generate the HTML file from the AsciiDoc file, install the ruby gem for asciidoctor
If you would like to generate the PDF file from the AsciiDoc file, install the ruby gem for asciidoctor-pdf
```
There's a function in the python file that runs a 'which asciidoctor' command. However, there's no path to the gems. Does the path need to be exported for this to work? Is this supposed to be ran in a virtual environment? I followed the getting started guide and everything installed correctly. The gems exist in the mscp_gems/ruby/2.6.0 directory. Not sure how to get this working.

**Comment by golbiga** [comment]
@FrankAllotta which branch are you running off of? If you're using `main` switch to one of the os branches.

**Reply by FrankAllotta** [reply]
I switched to the monterey branch and it worked.  Thank you!  Is there a reason why you have to switch branches? Is there some linkage that doesn't exist in the main branch. Trying to understand why/how this all works together.

**Comment by golbiga** [comment, marked as answer]
The `main` branch is only updated when a release is posted. Between that time we're making updates to the OS branches. So it's always best to work from the OS branch you're focusing on. I'll look to updating the Wiki and maybe putting something in the readme.

**Reply by FrankAllotta** [reply]
Thank you!
